WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift

```diff
@@ -287,7 +287,12 @@ private extension SiteStatsPeriodViewModel {
             mostRecentChartData.periodEndDate == periodSummary.periodEndDate {
             self.mostRecentChartData = periodSummary
         } else if let periodSummary = periodSummary, let chartData = mostRecentChartData, periodSummary.periodEndDate > chartData.periodEndDate {
-            mostRecentChartData = chartData
+            if let lastSummaryDataEntry = summaryData.last, periodSummary.periodEndDate == lastSummaryDataEntry.periodStartDate {
```

Comment:
I wonder if we could somehow improve the readability of all of these ifs. Without context, it takes a lot of time to understand these checks.

The easiest option I see is to make short comments above conditional statements. The harder option may be extracting the checks into methods.

WDYT?

Reply:
@staskus sounds good I made comments on the conditional statement that I touched which in my mind was the least transparent and most unclear on the intended behaviour